Clarify some code in do_create_interface_objects

pierrechevalier83 opened the pull request:
rval.get() is believed to be always null upon entering this function.
This assumption is verified by the added assertion.
It makes more sense to move the block of code that was moved inside
the if statement which is the only place where it can be initialized.

Fixes #7941.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/7943)

<!-- Reviewable:end -->

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

bors-servo commented:
:umbrella: The latest upstream changes (presumably #7960) made this pull request unmergeable. Please resolve the merge conflicts.

pierrechevalier83 committed:
Clarify some code in do_create_interface_objects

rval.get() is believed to be always null upon entering this function.
This assumption is verified by the added assertion.
It makes more sense to move the block of code that was moved inside
the if statement which is the only place where it can be initialized.

Fixes #7941.

Ms2ger commented:
Thanks, and sorry for the delay.
-S-awaiting-review -S-needs-rebase +S-awaiting-merge 
@bors-servo r+
<!-- Reviewable comment -K1-jeuHK_kJ8PZGvLVE:-715174275 -->

---

Reviewed 1 of 1 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7943)
<!-- Sent from Reviewable.io -->

bors-servo commented:
:pushpin: Commit a6a685e has been approved by `Ms2ger`

<!-- @bors-servo r=Ms2ger a6a685e9540f9cbd1167288472cf6d9d89475337 -->

bors-servo commented:
:hourglass: Testing commit a6a685e with merge 6111cf9...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/3484), [gonk](http://build.servo.org/builders/gonk/builds/2645), [linux-dev](http://build.servo.org/builders/linux-dev/builds/564), [linux-rel](http://build.servo.org/builders/linux-rel/builds/585), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/600), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/583), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/618)
